### PR TITLE
Improve debug logs

### PR DESCRIPTION
**Pull request by panleone**
## Abstract
debug logging imo is very important, but at the moment is quite unused since we have to wrap each console.log inside a `if(debug){...}`
This PR solves the issue by adding some util functions like `debugLog`;
DebugTopics are also added, so we can easily understand which part of the application printed the message
finally DebugTopics can be activated and deactivated: this can be done from the file `debug_config.json`

---

## Testing
Activate debug and see that it works as expected

### CODE REVIEW

File: scripts/utils.js [hunk 1]
```diff
@@ -137,3 +138,47 @@ export async function startBatch(
...
+/**
+ * Execute a given function if we are in debug mode
+ * @param {Function} func - a function we want to execute
+ * @param {...any} args - The arguments to pass to the function
+ */
+export function debugEval(func, ...args) {
+    if (debug) {
+        func(...args);
+    }
+}
```
Comment (Duddino) [inline scripts/utils.js:152]: I don't like this function, it's just going to lead to heisenbugs. Perhaps remove the `export`

**Review by Duddino — COMMENTED** [review]
It's also useful to have a way to disable certain topics. If I'm debugging network, I usually don't want logs for wallet for example. You can do this by putting the setting in a config file, (see chainparams.json). You can also add log error levels (Debug, warn, error) and a way to filter (use the console functions so we can also filter them using the dev tools).

**Comment by panleone** [comment]
ready for review

File: debug_config.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "debug_net": true,
+    "debug_global": true,
+    "debug_wallet": true,
+    "debug_mempool": true
```
[the remark this answers is not in the document]
Comment (panleone) [reply debug_config.json]: agree, I have modified

**Review by JSKitty — APPROVED** [review]
**tACK** 13652496bf24dd847fd461e9e9c7154f0cb3b06b

**Pull request merged**